build/yamls/antrea-agent-simulator.yaml

```diff
@@ -0,0 +1,225 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: antrea-agent-simulator
+  namespace: kube-system
+spec:
+  clusterIP: None
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    name: antrea-agent-simulator
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: antrea
+  name: antrea-agent-simulator
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    app: antrea
+  name: antrea-agent-simulator
+rules:
+- apiGroups:
+  - clusterinformation.antrea.tanzu.vmware.com
+  resources:
+  - antreaagentinfos
+  verbs:
+  - get
+  - create
+  - update
+  - delete
+- apiGroups:
+  - controlplane.antrea.tanzu.vmware.com
+  - networking.antrea.tanzu.vmware.com
+  resources:
+  - networkpolicies
+  - appliedtogroups
+  - addressgroups
+  verbs:
+  - get
+  - watch
+  - list
+- apiGroups:
+  - controlplane.antrea.tanzu.vmware.com
+  resources:
+  - nodestatssummaries
+  verbs:
+  - create
+- apiGroups:
+  - authentication.k8s.io
+  resources:
+  - tokenreviews
+  verbs:
+  - create
+- apiGroups:
+  - authorization.k8s.io
+  resources:
+  - subjectaccessreviews
+  verbs:
+  - create
+- apiGroups:
+  - ""
+  resourceNames:
+  - extension-apiserver-authentication
+  resources:
+  - configmaps
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resourceNames:
+  - antrea-ca
+  resources:
+  - configmaps
+  verbs:
+  - get
+  - watch
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    app: antrea
+  name: antrea-agent-simulator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: antrea-agent-simulator
+subjects:
+- kind: ServiceAccount
+  name: antrea-agent-simulator
+  namespace: kube-system
+---
+apiVersion: apps/v1
```

Comment:
We should also add `nodeAffinity` here to avoid simulators being scheduled to any quick ready simulate node.

Reply:
add label to the nodes? we do not know nodename until the cluster is ready.

Reply:
Not nodes but the simulator Pod, I think it's just like what we do for antrea Pods.

Reply:
OK, I think maybe it is better to remove the nodeAffinity from agent, and add to the simulator